blockly/src/server/Server.java

```diff
@@ -1270,7 +1270,8 @@ record EntityComponents(
     for (EntityComponents ec : entityComponents) {
       ec.vc.currentXVelocity(0);
       ec.vc.currentYVelocity(0);
-      ec.pc.position(ec.targetPosition.toCenteredPoint()); // snap to grid
+      // check the position-tile via new request in case a new level was loaded
+      ec.pc.position(Game.tileAT(ec.pc.position()).coordinate().toCenteredPoint()); // snap to grid
```

Comment:
Ich würde noch gerne eine if-Abfrage sehen ob auch TileAt wirkich ein Tile zurück gibt. 
```suggestion
     Tile endTile = Game.tileAT(ec.pc.position();
     if (endTile != null)
       ec.pc.position(endTile); // snap to grid
```
`.coordinate().toCenteredPoint())` wird dann durch #1821 obsolet